proparse/src/main/java/org/prorefactor/proparse/antlr4/PostLexer.java

```diff
@@ -78,25 +83,26 @@ public Token nextToken() throws TokenStreamException {
 
         }
       }
-    } catch (IOException | RecognitionException caught) {
-      throw new TokenStreamException(caught);
+    } catch (IOException | RecognitionException | TokenStreamException caught) {
+      throw new RuntimeException(caught);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00112)